src/ray/core_worker/reference_count.cc

```diff
@@ -1003,6 +1006,38 @@ absl::optional<LocalityData> ReferenceCounter::GetLocalityData(
   return locality_data;
 }
 
+void ReferenceCounter::PushToLocationSubscribers(ReferenceTable::iterator it) {
+  auto callbacks = it->second.location_subscription_callbacks;
+  it->second.location_subscription_callbacks.clear();
+  it->second.location_version++;
+  for (const auto callback : callbacks) {
+    callback(it->second.locations, it->second.object_size, it->second.location_version);
+  }
+}
+
+Status ReferenceCounter::GetObjectLocationsAsync(
```

Comment:
Maybe change the name to ListenObjectLocationChangeAsync?

Reply:
I've seen these kind of subscription callback registration functions called e.g. `GetAsync` elsewhere (memory store) which is why I went with the `Get{}Async` pattern, I was originally going to call this `SubscribeObjectLocations`. `Listen` suggests listening for an event rather than getting data or subscribing to data changes. How about `SubscribeObjectLocationsChange`? `Listen` or `Subscribe` implies async, so I think we can drop the `Async`.